tests/spec/features/save_and_return_module_spec.rb

```diff
@@ -1,107 +1,138 @@
 require 'capybara/rspec'
 require 'spec_helper'
 
+class SaveAndReturnApp < SitePrism::Page
+  set_url 'http://features-save-and-return-app:3000'
+
+  element :start_button, :button, 'Start'
+  element :continue_work_on_a_saved_form_button, 'p a[href="/return"]'
+  element :save_and_come_back_later_button, 'button[name="setupReturn"]'
+  element :continue_button, :button, 'Continue'
+  element :email_field, :field, 'Email address'
+  element :email_error_summary, '.govuk-error-summary a[href="#return_start_email"]'
+  element :email_error_message, '.govuk-error-message'
+  element :continue_with_this_form_button, :link, 'Continue with this form'
+  element :back_link, :link, 'Back'
+
+  element :usn_field, :field, 'Unique submission number (USN)'
+
+  def has_sign_in_link?
+    page.has_selector? 'h1', text: 'Get a sign-in link'
+  end
+
+  def has_email_message?(message)
+    page.has_selector?('h1 + p', text: message)
+  end
+end
+
 describe 'Using Save and Return' do
-  it 'puts a link onto the start page' do
-    visit 'http://features-save-and-return-app:3000'
+  let(:form) { SaveAndReturnApp.new }
 
-    # Save and Return "Continue work on a saved form"
-    expect(page).to have_selector 'p a[href="/return"]', text: 'Continue work on a saved form'
+  before { form.load }
+
+  it 'puts a link onto the start page' do
+    expect(form).to have_continue_work_on_a_saved_form_button
+    expect(
+      form.continue_work_on_a_saved_form_button.text
+    ).to eq('Continue work on a saved form')
   end
 
   it 'puts a save and return button into the form' do
-    visit 'http://features-save-and-return-app:3000'
-    click_on 'Start'
-
-    expect(page).to have_selector 'button[name="setupReturn"]', text: 'Save and come back later'
+    form.start_button.click
+    expect(form).to have_save_and_come_back_later_button
+    expect(
+      form.save_and_come_back_later_button.text
+    ).to eq('Save and come back later')
   end
 
   it 'puts a save and return message into the page' do
-    visit 'http://features-save-and-return-app:3000'
-    click_on 'Start'
-
-    expect(page).to have_selector 'p', text: /Your session will time out after \d+ minutes of inactivity/
-    expect(page).to have_selector 'p', text: 'Save your details if you want to come back later'
+    form.start_button.click
+    expect(form).to have_content(
+      /Your session will time out after \d+ minutes of inactivity/
```

Comment:
Do we need this? Shouldn't we test this in its own spec?

Reply:
I don't know. This was already there, I just adapted. Should I delete then?

Reply:
Hm, I must have written it, then

Kill it

Reply:
Actually, I remember writing the `\d` part. You can kill it